conjure-java-core/src/main/java/com/palantir/conjure/java/types/UnionGenerator.java

```diff
@@ -574,6 +579,19 @@ private static TypeSpec generateBase(
                 .collect(Collectors.toList());
     }
 
+    private static AnnotationSpec wrapperConstructorParameterAnnotation(FieldDefinition field) {
+        AnnotationSpec.Builder builder = AnnotationSpec.builder(JsonSetter.class)
+                .addMember("value", "$S", field.getFieldName().get());
+        if (isListSetOrMap(field.getType())) {
+            builder.addMember("nulls", "$T.AS_EMPTY", Nulls.class);
+        }
+        return builder.build();
+    }
+
+    private static boolean isListSetOrMap(Type type) {
+        return type.accept(TypeVisitor.IS_LIST) || type.accept(TypeVisitor.IS_SET) || type.accept(TypeVisitor.IS_MAP);
```

Comment:
teeny bit surprising that this includes 3/4 container types, i.e. doesn't include optionals??

Reply:
No need to include optional here because null is already deserialized into an empty optional. I added test coverage for the null and omitted cases in this PR :-)